### PR TITLE
non-aloc-concurent version

### PR DESCRIPTION
**Pull request by MatejMagat305**

### CODE REVIEW

**Comment by andydotxyz** [comment]
Please provide some more information, this title does not make sense.

**Comment by MatejMagat305** [comment]
improvements #1

File: AUTHORS [hunk 1]
```diff
@@ -1 +1 @@
-Andy Williams <andy@andy.xyz>
+Andy Williams <andy@andy.xyz>
```
Comment (andydotxyz) [inline AUTHORS:1]: Files should have newlines at the end - I don't think this change is needed

File: LICENSE [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (andydotxyz) [inline LICENSE:1]: Please don't edit our license file

File: board.go [hunk 3]
```diff
@@ -2,6 +2,8 @@ package main
 
 type board struct {
 	cells         [][]bool
+	nextGenCells         [][]bool
```
Comment (andydotxyz) [inline board.go:5]: This formatting looks off, please run `goimports -w .` to correct.

File: board.go [hunk 4]
```diff
@@ -39,42 +41,39 @@ func (b *board) countNeighbours(x, y int) int {
...
+	for y := 0; y < height; y++ {
+		for x := 0; x < width; x++ {
+			go func(x,y int) {
+				n := b.countNeighbours(x, y)
+				if b.cells[y][x] {
+					b.nextGenCells[y][x] = n == 2 || n == 3
+				} else {
+					b.nextGenCells[y][x] = n == 3
+				}
+				b.ch<-true
```
Comment (andydotxyz) [inline board.go:60]: I think this is local to the method, so does it need to be in the game state?
Also I wonder, does it not fit the `WorkGroup` model better?

**Review by andydotxyz — CHANGES_REQUESTED** [review]
A question about the wait model, and some notes about un-needed changes that should be backed out.

**Comment by Jacalz** [comment]
I did some benchmarking on this and it seems that the concurrent version is slower than the version before (tested using `WorkGroup` instead of channels). The overhead of spinning up goroutines is larger than the benefit of concurrency. It seems like the pre-allocated board does improve quite a bit though.

@MatejMagat305 Do you wish to  update your PR with that and try to address the other review as well? I can open a PR if you don't wish to continue this.

**Comment by MatejMagat305** [comment]
yes, sorry, the only impovement is not aloc ...

**Pull request closed without merging**